### ISSUE TITLE
Typescript support for jest?

### ISSUE THREAD

**Issue by andrehadianto** [react-native-async-storage/async-storage#504]
As the title implies, is there typescript support for `@react-native-async-storage/async-storage`?

I figured in other discussions that the `@types` should be the same as the original `react-native` async-storage or the predecessor `react-native-community` async-storage. However, I don't know how to implement the previous types on this module.

Instructions would be appreciated!

```
Could not find a declaration file for module '@react-native-async-storage/async-storage/jest/async-storage-mock'
```

**Comment by invm** [comment]
What I did was to go to 
```node_modules/@react-native-async-storage/async-storage/types/index.d.ts```

copied the whole block down,  and changed the name to:
```declare module '@react-native-async-storage/async-storage/jest/async-storage-mock' {...}```

I do not know if it is the right approach, if it is, someone could do a PR and solve the issue for the next person.

**Comment by JoeKlimcak** [comment]
Would also like this supported. When I follow your [jest-setup](https://react-native-async-storage.github.io/async-storage/docs/advanced/jest/) in my TS project, I get:

> [@RNC/AsyncStorage]: NativeModule: AsyncStorage is null.

**Comment by invm** [comment]
@JoeKlimcak 
Did you try doing what I did?

**Comment by Krizzu** [comment]
Typescript issue comes from the fact that the mock is not declared in our typings. In that case, I advise to use [Jest setup file to mock AsyncStorage](https://react-native-async-storage.github.io/async-storage/docs/advanced/jest/#with-jest-setup-file)

**Comment by tido64** [comment]
Closing due to inactivity. If you think this is a mistake, please reply and we'll reopen it.

**Comment by Guicbdiniz** [comment]
> Closing due to inactivity. If you think this is a mistake, please reply and we'll reopen it.

I'm having the same issue here with the import.
Although I'm able to import `import AsyncStorage from "@react-native-async-storage/async-storage"`, I get a type error when importing `import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock'`

**Comment by MatkoMilic** [comment]
Having the same issue as the comment saying: I'm having the same issue here with the import.
Although I'm able to import import AsyncStorage from "@react-native-async-storage/async-storage", I get a type error when importing import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock'

**Comment by bytehala** [comment]
This shouldn't have been closed. The mocks are still not typed. Converting your jest setup from JS to TS will yield the error everyone is talking about.

**Comment by tido64** [comment]
I think this might be addressed by #805. Please test it and let me know how it goes.

**Comment by bytehala** [comment]
@tido64 I'm taking a look now.

**Comment by jkdowdle** [comment]
> I think this might be addressed by #805. Please test it and let me know how it goes.

Updating appeared to solve the issue for me
